test/suites/clustering.sh

```diff
@@ -3090,42 +3091,83 @@ test_clustering_rebalance() {
   LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node2 | grep -xF "status: Offline"
   LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node2 | grep -xF "database: false"
   LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node4 | grep -xF "status: Online"
   LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node4 | grep -F -- "- database"
 
-  # Respawn the second node. It won't be able to disrupt the current leader,
-  # since dqlite uses pre-vote.
+  echo "Respawn the second node."
+  # It won't be able to disrupt the current leader, since dqlite uses pre-vote.
   respawn_lxd_cluster_member "${ns2}" "${LXD_TWO_DIR}"
   sleep 12
 
+  echo "Verify the second node is back online and has database role."
   LXD_DIR="${LXD_ONE_DIR}" lxc cluster list
   LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node2 | grep -xF "status: Online"
   LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node2 | grep -xF "database: true"
 
+  echo "Add a fifth node to the cluster."
+  # Spawn a fifth node
+  setup_clustering_netns 5
+  LXD_FIVE_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  ns5="${prefix}5"
+  spawn_lxd_and_join_cluster "${ns5}" "${bridge}" "${cert}" 5 1 "${LXD_FIVE_DIR}" "${LXD_ONE_DIR}"
+
+  echo "Remove standby role and add client role to fifth node."
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster role remove node5 database-standby
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster role add node5 database-client
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster list # For debugging.
+  [ "$(LXD_DIR="${LXD_ONE_DIR}" lxc cluster list | grep -Fc "database-client")" = "1" ]
+
+  echo "Wait for rebalancing to demote node5 to spare (non-database member)."
+  sleep 15
+
+  echo "Verify database-standby role has been removed from node5."
+  [ "$(LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node5 | grep -Fc "database-standby")" = "0" ]
+
+  echo "Verify node5 has spare raft role (non-database member)."
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node5 | grep -xF "database: false"
+
+  echo "Kill node4 to trigger rebalancing."
+  kill -9 "$(< "${LXD_FOUR_DIR}/lxd.pid")"
+  sleep 15
+
+  echo "Verify node4 is offline and has been demoted."
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node4 | grep -xF "status: Offline"
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node4 | grep -xF "database: false"
+
+  echo "Verify node5 with database-client role was not promoted during rebalancing."
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node5 | grep -xF "database: false"
+
+  echo "Verify node2 (standby without database-client) was promoted to voter instead."
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node2 | grep -xF "database: true"
+  [ "$(LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node2 | grep -cF -- "- database-standby")" = "0" ]
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster list # For debugging.
+
+  echo "Clean up"
   LXD_DIR="${LXD_ONE_DIR}" lxd shutdown
   LXD_DIR="${LXD_TWO_DIR}" lxd shutdown
   LXD_DIR="${LXD_THREE_DIR}" lxd shutdown
-  LXD_DIR="${LXD_FOUR_DIR}" lxd shutdown
```

Comment:
I'd keep that line but commented out with an additional note that `LXD_FOUR_DIR` was abruptly killed earlier.